From fd00fe0349492a48774e5a72f7c8b931cdf6e655 Mon Sep 17 00:00:00 2001
From: Joe McGill <jmcgill@us.ibm.com>
Date: Mon, 7 Nov 2016 23:04:49 -0600
Subject: [PATCH] p9.fbc.ab_hp.scom.initfile -- update X/A cmd rate settings

  consider aggregate mode programming
  update to use mesh frequency rather than bus frequency

Change-Id: I6e4b85940a7977296d1fae6fbed5c1e7b84824d5
---
 .../procedures/hwp/initfiles/p9_fbc_ab_hp_scom.C   | 107 +++++++++++++++------
 1 file changed, 78 insertions(+), 29 deletions(-)

diff --git a/src/import/chips/p9/procedures/hwp/initfiles/p9_fbc_ab_hp_scom.C b/src/import/chips/p9/procedures/hwp/initfiles/p9_fbc_ab_hp_scom.C
index b167d9e..ab2ff39 100644
--- a/src/import/chips/p9/procedures/hwp/initfiles/p9_fbc_ab_hp_scom.C
+++ b/src/import/chips/p9/procedures/hwp/initfiles/p9_fbc_ab_hp_scom.C
@@ -46,9 +46,8 @@ constexpr auto literal_12 = 12;
 constexpr auto literal_13 = 13;
 constexpr auto literal_14 = 14;
 constexpr auto literal_15 = 15;
-constexpr auto literal_1600 = 1600;
-constexpr auto literal_16 = 16;
-constexpr auto literal_2577 = 2577;
+constexpr auto literal_2000 = 2000;
+constexpr auto literal_1611 = 1611;
 
 fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_CHIP>& TGT0,
                                     const fapi2::Target<fapi2::TARGET_TYPE_SYSTEM>& TGT1)
@@ -185,8 +184,8 @@ fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_
             break;
         }
 
-        auto l_def_A_CMD_RATE_4B_R = (((literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_2577) % (((
-                                          l_TGT1_ATTR_FREQ_A_MHZ * literal_16) * literal_1600) * literal_2));
+        auto l_def_A_CMD_RATE_4B_R = (((literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611) %
+                                      (l_TGT1_ATTR_FREQ_A_MHZ * literal_2000));
         fapi2::ATTR_PROC_FABRIC_A_BUS_WIDTH_Type l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH;
         l_rc = FAPI_ATTR_GET(fapi2::ATTR_PROC_FABRIC_A_BUS_WIDTH, TGT1, l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH);
 
@@ -196,11 +195,17 @@ fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_
             break;
         }
 
-        auto l_def_A_CMD_RATE_D = (((l_TGT1_ATTR_FREQ_A_MHZ * literal_16) * literal_1600) * literal_2);
-        auto l_def_A_CMD_RATE_4B_N = ((literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_2577);
-        auto l_def_A_CMD_RATE_2B_R = (((literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_2577) % (((
-                                          l_TGT1_ATTR_FREQ_A_MHZ * literal_16) * literal_1600) * literal_2));
-        auto l_def_A_CMD_RATE_2B_N = ((literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_2577);
+        auto l_def_A_CMD_RATE_D = (l_TGT1_ATTR_FREQ_A_MHZ * literal_2000);
+        auto l_def_A_CMD_RATE_4B_N = ((literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611);
+        auto l_def_A_CMD_RATE_4B_RA = (((literal_5 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611) %
+                                       (l_TGT1_ATTR_FREQ_A_MHZ * literal_2000));
+        auto l_def_A_CMD_RATE_4B_NA = ((literal_5 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611);
+        auto l_def_A_CMD_RATE_2B_R = (((literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611) %
+                                      (l_TGT1_ATTR_FREQ_A_MHZ * literal_2000));
+        auto l_def_A_CMD_RATE_2B_N = ((literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611);
+        auto l_def_A_CMD_RATE_2B_RA = (((literal_10 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611) %
+                                       (l_TGT1_ATTR_FREQ_A_MHZ * literal_2000));
+        auto l_def_A_CMD_RATE_2B_NA = ((literal_10 * l_TGT1_ATTR_FREQ_PB_MHZ) * literal_1611);
         {
             l_rc = fapi2::getScom( TGT0, 0x501180bull, l_scom_buffer );
 
@@ -784,26 +789,46 @@ fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_
             }
 
             {
-                if (((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_4_BYTE)
-                     && (l_def_A_CMD_RATE_4B_R != literal_0)))
+                if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_4_BYTE)
+                      && (l_def_LINK_A_AGGREGATE_EN == literal_0)) && (l_def_A_CMD_RATE_4B_R != literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> ((l_def_A_CMD_RATE_4B_N / l_def_A_CMD_RATE_D), 56, 8, 56 );
                 }
-                else if (((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_4_BYTE)
-                          && (l_def_A_CMD_RATE_4B_R == literal_0)))
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_4_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_0)) && (l_def_A_CMD_RATE_4B_R == literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> (((l_def_A_CMD_RATE_4B_N / l_def_A_CMD_RATE_D) - literal_1), 56, 8, 56 );
                 }
-                else if (((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_2_BYTE)
-                          && (l_def_A_CMD_RATE_2B_R != literal_0)))
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_4_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_1)) && (l_def_A_CMD_RATE_4B_RA != literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> ((l_def_A_CMD_RATE_4B_NA / l_def_A_CMD_RATE_D), 56, 8, 56 );
+                }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_4_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_1)) && (l_def_A_CMD_RATE_4B_RA == literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> (((l_def_A_CMD_RATE_4B_NA / l_def_A_CMD_RATE_D) - literal_1), 56, 8, 56 );
+                }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_0)) && (l_def_A_CMD_RATE_2B_R != literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> ((l_def_A_CMD_RATE_2B_N / l_def_A_CMD_RATE_D), 56, 8, 56 );
                 }
-                else if (((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_2_BYTE)
-                          && (l_def_A_CMD_RATE_2B_R == literal_0)))
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_0)) && (l_def_A_CMD_RATE_2B_R == literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> (((l_def_A_CMD_RATE_2B_N / l_def_A_CMD_RATE_D) - literal_1), 56, 8, 56 );
                 }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_1)) && (l_def_A_CMD_RATE_2B_RA != literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> ((l_def_A_CMD_RATE_2B_NA / l_def_A_CMD_RATE_D), 56, 8, 56 );
+                }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_A_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_A_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_A_AGGREGATE_EN == literal_1)) && (l_def_A_CMD_RATE_2B_RA == literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> (((l_def_A_CMD_RATE_2B_NA / l_def_A_CMD_RATE_D) - literal_1), 56, 8, 56 );
+                }
             }
 
             l_rc = fapi2::putScom(TGT0, 0x501180bull, l_scom_buffer);
@@ -860,7 +885,7 @@ fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_
             break;
         }
 
-        auto l_def_X_CMD_RATE_4B_R = ((literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ) % (l_TGT1_ATTR_FREQ_X_MHZ * literal_8));
+        auto l_def_X_CMD_RATE_4B_R = ((literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ) % l_TGT1_ATTR_FREQ_X_MHZ);
         fapi2::ATTR_PROC_FABRIC_X_BUS_WIDTH_Type l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH;
         l_rc = FAPI_ATTR_GET(fapi2::ATTR_PROC_FABRIC_X_BUS_WIDTH, TGT1, l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH);
 
@@ -870,10 +895,14 @@ fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_
             break;
         }
 
-        auto l_def_X_CMD_RATE_D = (l_TGT1_ATTR_FREQ_X_MHZ * literal_8);
+        auto l_def_X_CMD_RATE_D = l_TGT1_ATTR_FREQ_X_MHZ;
         auto l_def_X_CMD_RATE_4B_N = (literal_6 * l_TGT1_ATTR_FREQ_PB_MHZ);
-        auto l_def_X_CMD_RATE_2B_R = ((literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ) % (l_TGT1_ATTR_FREQ_X_MHZ * literal_8));
+        auto l_def_X_CMD_RATE_4B_RA = ((literal_5 * l_TGT1_ATTR_FREQ_PB_MHZ) % l_TGT1_ATTR_FREQ_X_MHZ);
+        auto l_def_X_CMD_RATE_4B_NA = (literal_5 * l_TGT1_ATTR_FREQ_PB_MHZ);
+        auto l_def_X_CMD_RATE_2B_R = ((literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ) % l_TGT1_ATTR_FREQ_X_MHZ);
         auto l_def_X_CMD_RATE_2B_N = (literal_12 * l_TGT1_ATTR_FREQ_PB_MHZ);
+        auto l_def_X_CMD_RATE_2B_RA = ((literal_10 * l_TGT1_ATTR_FREQ_PB_MHZ) % l_TGT1_ATTR_FREQ_X_MHZ);
+        auto l_def_X_CMD_RATE_2B_NA = (literal_10 * l_TGT1_ATTR_FREQ_PB_MHZ);
         {
             l_rc = fapi2::getScom( TGT0, 0x501180full, l_scom_buffer );
 
@@ -1322,26 +1351,46 @@ fapi2::ReturnCode p9_fbc_ab_hp_scom(const fapi2::Target<fapi2::TARGET_TYPE_PROC_
             }
 
             {
-                if (((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_4_BYTE)
-                     && (l_def_X_CMD_RATE_4B_R != literal_0)))
+                if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_4_BYTE)
+                      && (l_def_LINK_X_AGGREGATE_EN == literal_0)) && (l_def_X_CMD_RATE_4B_R != literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> ((l_def_X_CMD_RATE_4B_N / l_def_X_CMD_RATE_D), 56, 8, 56 );
                 }
-                else if (((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_4_BYTE)
-                          && (l_def_X_CMD_RATE_4B_R == literal_0)))
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_4_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_0)) && (l_def_X_CMD_RATE_4B_R == literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> (((l_def_X_CMD_RATE_4B_N / l_def_X_CMD_RATE_D) - literal_1), 56, 8, 56 );
                 }
-                else if (((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_2_BYTE)
-                          && (l_def_X_CMD_RATE_2B_R != literal_0)))
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_4_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_1)) && (l_def_X_CMD_RATE_4B_RA != literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> (((l_def_X_CMD_RATE_4B_NA / l_def_X_CMD_RATE_D) + literal_1), 56, 8, 56 );
+                }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_4_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_1)) && (l_def_X_CMD_RATE_4B_RA == literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> ((l_def_X_CMD_RATE_4B_NA / l_def_X_CMD_RATE_D), 56, 8, 56 );
+                }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_0)) && (l_def_X_CMD_RATE_2B_R != literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> ((l_def_X_CMD_RATE_2B_N / l_def_X_CMD_RATE_D), 56, 8, 56 );
                 }
-                else if (((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_2_BYTE)
-                          && (l_def_X_CMD_RATE_2B_R == literal_0)))
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_0)) && (l_def_X_CMD_RATE_2B_R == literal_0)))
                 {
                     l_scom_buffer.insert<uint64_t> (((l_def_X_CMD_RATE_2B_N / l_def_X_CMD_RATE_D) - literal_1), 56, 8, 56 );
                 }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_1)) && (l_def_X_CMD_RATE_2B_RA != literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> ((l_def_X_CMD_RATE_2B_NA / l_def_X_CMD_RATE_D), 56, 8, 56 );
+                }
+                else if ((((l_TGT1_ATTR_PROC_FABRIC_X_BUS_WIDTH == fapi2::ENUM_ATTR_PROC_FABRIC_X_BUS_WIDTH_2_BYTE)
+                           && (l_def_LINK_X_AGGREGATE_EN == literal_1)) && (l_def_X_CMD_RATE_2B_RA == literal_0)))
+                {
+                    l_scom_buffer.insert<uint64_t> (((l_def_X_CMD_RATE_2B_NA / l_def_X_CMD_RATE_D) - literal_1), 56, 8, 56 );
+                }
             }
 
             l_rc = fapi2::putScom(TGT0, 0x501180full, l_scom_buffer);
-- 
1.8.2.2

